docs/cli-usage.md

```diff
@@ -150,6 +150,46 @@ Identifier styles (ex: `FooBar`, `fooBar`, `foo_bar`, `FOO_BAR`, `m_fooBar`)
 | `--ident-objc-type-param ...` |
 | `--ident-objc-local ...` |
 | `--ident-objc-file ...` |
+|   |
 
+The _Identifier Style_ is a text conversion hint on how to generate names from the djinni idl file.
 
+Example:
+The djinni idl for an enum
 
+```
+SomeEnum = enum {
+    Value;
+}
+```
+
+transforms per default to the following Java code
+
+```
+public enum SomeEnum {
+    VALUE,
+    ;
+}
+```
+As you see, _VALUE_ is now in upper case letters.
+
+If you use `--ident-java-enum foo_bar` then _Value_ will not be transformed into upper case and the following Java code will be generated.
+
+```
+public enum SomeEnum {
+    Value,
+    ;
+}
+```
+
+This way you can adopt code generation to some extend to your existing coding style.
+
+
+The best way to find out how that works is playing around with different values for different identifier options.
+
+**If you wish not text transformation to happen, use the `foo_bar` style.**
+
+!!! note
+
+    Some defaults have been set while back, that, for example enum names are transformed to uppercase letters.
+    At the moment we do not want to change that behavior, but please be aware that we think the default should always be no text transformation for all settings.
```

Comment:
I have trouble understanding what the first sentence wants to say. Additionally I think we should give a reason for why we recommend to disable text transformation.

I propose a note like this:

```suggestion
    Some types do have text transformation enabled by default. To avoid breaking existing builds, this will not be changed in the future.
    Nevertheless it is recommended to disable all text transformation to avoid issues in the target language.
```

Am I right with the reason for our recommendation? I did not follow the discussion on this in detail.

Reply:
its not just types that have transformations, but also enum values, member fields .....

yes, we can improve the second sentence,  I would like to transport the message that in future that might change
so please start setting everything explicit now, or so ... just to open a door

Reply:
I will think a little later about how that could be improved, if you have further suggestion, I am happy to merge them in

Reply:
```suggestion
    Some identifiers do have text transformation enabled by default. 
    Nevertheless make sure to **explicitly set** text transformation styles if you need them! All text transformation may be disabled by default in a future release of the generator.
```